packages/experience/src/containers/VerificationCode/use-register-flow-code-verification.ts

```diff
@@ -56,7 +56,10 @@ const useRegisterFlowCodeVerification = (
     const { type, value } = identifier;
 
     // Should not redirect user to sign-in if is register-only mode
-    if (signInMode === SignInMode.Register) {
+    if (
+      signInMode === SignInMode.Register ||
+      !signInMethods.find(({ identifier }) => identifier === type)?.verificationCode
```

Comment:
better to update comment for why this logic is added

Reply:
updated